### PR TITLE
Active-Active Region Application Failover

### PR DESCRIPTION
**Pull request by AlonMagrafta**
Yossi and Yuval (Inbar) wrote this doc.
We still need to upload a python script as stated in line 59.
@bmansheim let's coordinate before going live with that one. Thanks

### CODE REVIEW

**Comment by bmansheim** [comment]
@AlonMagrafta I finished the style review. We just need the Python script in order to be ready to merge, right?

**Comment by bmansheim** [comment]
@banker I added the script to the branch but couldn't add it to the page in a way to download nicely.

**Comment by kaitlynmichaelredislabs** [comment]
@AlonMagrafta is this something we need to complete before 6.0.20?

**Comment by AlonMagrafta** [comment]
@kaitlynmichael we can do this after 6.0.20, and once it's ready merge and publish, it's not release related

File: content/rs/references/developing-for-active-active/region-application-failover-active-active.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+The most reliable method for health-checking replication is by using the Redis publish/subscribe (pub/sub) mechanism.
+
+{{< note >}}
+Note that this document does not suggest that Redis pub/sub is reliable in the common sense. Messages can get lost in certain conditions, but that is acceptable in this case because typically the application determines that replication is down only after not being able to deliver a number of messages over a period of time.
+{{< /note >}}
+
+When you use the pub/sub data type to detect failures:
+
+1. The application establishes a connection to all replicas and subscribes to a dedicated per-replica channel.
```
Comment (kaitlynmichael) [inline content/rs/references/developing-for-active-active/region-application-failover-active-active.md:43]:
```suggestion
1. Connects to all replicas and subscribes to a dedicated channel for each replica.
```

File: content/rs/references/developing-for-active-active/region-application-failover-active-active.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Replication failures are more difficult to detect reliably without causing false positives. Replication failures can include: network split, replication configuration issues, remote replica failures.
+
+The most reliable method for health-checking replication is by using the Redis publish/subscribe (pub/sub) mechanism.
+
+{{< note >}}
+Note that this document does not suggest that Redis pub/sub is reliable in the common sense. Messages can get lost in certain conditions, but that is acceptable in this case because typically the application determines that replication is down only after not being able to deliver a number of messages over a period of time.
+{{< /note >}}
+
+When you use the pub/sub data type to detect failures:
```
Comment (kaitlynmichael) [inline content/rs/references/developing-for-active-active/region-application-failover-active-active.md:41]:
```suggestion
When you use the pub/sub data type to detect failures, the application:
```
```suggestion
When you use the pub/sub data type to detect failures:
```

File: content/rs/references/developing-for-active-active/region-application-failover-active-active.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+1. The application establishes a connection to all replicas and subscribes to a dedicated per-replica channel.
+1. The application establishes another connection to all replicas and periodically publishes a uniquely identifiable message.
```
Comment (kaitlynmichael) [inline content/rs/references/developing-for-active-active/region-application-failover-active-active.md:44]:
```suggestion
1. Connects to all replicas and periodically publishes a uniquely identifiable message.
```
```suggestion
1. The application establishes another connection to all replicas and periodically publishes a uniquely identifiable message.
```

File: content/rs/references/developing-for-active-active/region-application-failover-active-active.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+1. The application monitors received messages and ensures that it is able to receive its own messages within a predetermined window of time.
```
Comment (kaitlynmichael) [inline content/rs/references/developing-for-active-active/region-application-failover-active-active.md:45]:
```suggestion
1. Monitors received messages and ensures that it is able to receive its own messages within a predetermined window of time.
```
```suggestion
1. The application monitors received messages and ensures that it is able to receive its own messages within a predetermined window of time.
```

File: content/rs/references/developing-for-active-active/region-application-failover-active-active.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+You can also rely on known dataset changes to monitor the reliability of the replication stream,
```
Comment (kaitlynmichael) [inline content/rs/references/developing-for-active-active/region-application-failover-active-active.md:47]:
```suggestion
You can also use known dataset changes to monitor the reliability of the replication stream,
```
```suggestion
You can also rely on known dataset changes to monitor the reliability of the replication stream,
```

**Review by kaitlynmichael — COMMENTED** [review]
Reveiwed up to line 50

**Review by kaitlynmichael — APPROVED** [review]

**Pull request merged**